### ISSUE TITLE
[Bug] `data` for POST request is optional

### ISSUE THREAD

**Issue by QzCurious** [ecyrbe/zodios#171]
I've define a body parameter for my post request. But I can ignore `data` without ts showing error.

[codesandbox](https://codesandbox.io/s/cold-sun-57ky11?file=/src/api.ts)

```ts
import { makeApi, Zodios } from "@zodios/core";
import { z } from "zod";

export const Request = z.object({
  page: z.number().int().min(1),
  perpage: z.number().int().max(20)
});

const Response = z.object({});

const userApi = makeApi([
  {
    method: "post",
    path: "user/list",
    alias: "userList",
    immutable: true,
    parameters: [
      {
        name: "body",
        type: "Body",
        schema: Request
      }
    ],
    response: Response
  }
]);

const apiClient = new Zodios(userApi);
apiClient.userList(); // <-- data should be required
apiClient.post("user/list"); // <-- data should be required

```

**Comment by ecyrbe** [comment]
Thank you for opening this issue,
I'll check this bug this weekend.

**Comment by thelinuxlich** [comment]
I need to investigate more but I'm using a library to extract [err, result] from async ops and specifically with zod POST endpoints it's returning a different type, forcing me to use `as` assertion

**Comment by thelinuxlich** [comment]
Now, thinking about it again, wouldn't it be better if the POST endpoints instead of accepting body as the first parameter, did like the GET endpoints with a single object(so body would be a property of this object)?

**Comment by ecyrbe** [comment]
yes, this is maybe a disgn mistake on my part. it makes things harder to implement and also harder to autocomplete.
i'll think about it. i don't want to create a breaking change, so i need to try if overload could work here to support both.

In the mean time i'll try to first fix the reported bug.

**Comment by ecyrbe** [comment]
Hello @QzCurious,

Just to keep you informed that i found a way to handle this. it's not really pretty, but works.
This issue was really hard to solve because we can't declare overloads to solve this issue on all parameters types (issue is appearing for query, params and header parameters, not just body.  
I need a couple more days to finalize the change on all impacted packages (core, plugins, react; solid)

**Comment by ecyrbe** [comment]
fixed in `v10.2.0` : 
- `@zodios/core`
- `@zodios/plugins`
- `@zodios/react`
- `@zodios/solid`
